### PR TITLE
Ux improvements

### PR DESCRIPTION
**Pull request by ClementPasteau**
- some wording changes
- create project quickly without the examples dialog
- add the recommended templates at the top of the build section

image
Capture d’écran 2022-08-24 à 15 16 16
Capture d’écran 2022-08-24 à 15 16 38

https://user-images.githubusercontent.com/4895034/186428786-6e3b51aa-bf12-484b-8420-4ee44d0fbab2.mov

### CODE REVIEW

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/ExamplesGrid.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    : exampleShortHeaders;
+
+  return (
+    <div style={styles.examplesContainer}>
+      <Line noMargin>
+        <GridList
+          cols={getColumnsFromWidth(windowWidth)}
+          style={styles.grid}
+          cellHeight="auto"
+          spacing={16}
```
Comment (AlexandreSi) [inline newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/ExamplesGrid.js:100]: Should this value be at the top of the file and linked to other values? (like the 8 I see in the grid max width)

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/ExamplesLine.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import ExamplesGrid from './ExamplesGrid';
+import { prepareExamples } from '../../../../AssetStore/ExampleStore';
+
+type ExamplesLineProps = {|
+  exampleShortHeaders: Array<ExampleShortHeader>,
+  onExpand: () => void,
+  onOpen: (exampleShortHeader: ExampleShortHeader) => void,
+|};
+
+const ExamplesLine = ({
```
Comment (AlexandreSi) [inline newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/ExamplesLine.js:19]: I'm not sure if I'm right or if I'm bad at english but `ExamplesRow` sound better to me.

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/index.js [hunk 3]
```diff
@@ -261,6 +283,29 @@ const BuildSection = React.forwardRef<Props, BuildSectionInterface>(
...
+
+        projectMetadata = await onCreateBlank({
+          i18n,
+          settings,
+        });
+
+        if (!projectMetadata) return;
+
+        setPreCreationDialogOpen(false);
+        onOpenProjectAfterCreation({ ...projectMetadata });
```
Comment (AlexandreSi) [inline newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/index.js:303]: `projectMetadata` only should work I guess

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/index.js [hunk 4]
```diff
@@ -189,8 +187,13 @@ export const HomePage = React.memo<Props>(
                       ref={buildSectionRef}
                       project={project}
                       canOpen={canOpen}
-                      onOpen={onOpen}
-                      onCreateProject={onCreateProject}
+                      onChooseProject={onChooseProject}
+                      onCreateBlank={onCreateBlank}
+                      onShowAllExamples={() => onCreateProject(null)}
```
Comment (AlexandreSi) [inline newIDE/app/src/MainFrame/EditorContainers/HomePage/index.js:192]: This seems weird, maybe just a naming issue

File: newIDE/app/src/GameDashboard/GameCard.js [hunk 5]
```diff
@@ -49,32 +49,34 @@ type TogglableProperties =
...
-        Liluo.io, to let any user give you feedback. Continue?
+        You are about to activate a feedback banner on your Liluo.io game page.
+        By doing this you will receive feedback from any Liluo.io visitor.
+        Do you want to continue?
       `,
     false: t`
-        You are about to de-activate the feedback banner on your game page on
-        Liluo.io. Continue ?
+        You are about to de-activate the feedback banner on your Liluo.io game page.
+        Do you want to Continue ?
```
Comment (4ian) [inline newIDE/app/src/GameDashboard/GameCard.js:79]:
```suggestion
        Do you want to continue ?
```

File: newIDE/app/src/GameDashboard/GameCard.js [hunk 6]
```diff
@@ -49,32 +49,34 @@ type TogglableProperties =
...
-        Liluo.io, asking for feedback on your game. Continue?
+        You are about to activate a feedback banner on all builds of this game.
+        By doing this you're allowing feedback from any player who has access to your Liluo.io build URLs.
+        Do you want to continue?
       `,
     false: t`
-        You are about to de-activate the feedback banner on all your build pages
-        on Liluo.io. Continue ?
+        You are about to de-activate the feedback banner on all your Liluo.io build pages.
+        Do you want to Continue ?
```
Comment (4ian) [inline newIDE/app/src/GameDashboard/GameCard.js:68]:
```suggestion
        Do you want to continue ?
```

File: newIDE/app/src/GameDashboard/GameCard.js [hunk 7]
```diff
@@ -49,32 +49,34 @@ type TogglableProperties =
 const confirmationMessage = {
   discoverable: {
     true: t`
-        You are about to make this game discoverable on categories pages on
-        Liluo.io. Continue?
+        You are about to make this game discoverable on Liluo.io categories pages. 
+        Do you want to Continue?
```
Comment (4ian) [inline newIDE/app/src/GameDashboard/GameCard.js:53]:
```suggestion
        Do you want to continue?
```

File: newIDE/app/src/GameDashboard/GameCard.js [hunk 8]
```diff
@@ -49,32 +49,34 @@ type TogglableProperties =
...
       `,
     false: t`
-        You are about to hide this game from categories pages on Liluo.io.
-        Continue?
+        You are about to hide this game from Liluo.io categories pages.
+        Do you want to Continue?
```
Comment (4ian) [inline newIDE/app/src/GameDashboard/GameCard.js:57]:
```suggestion
        Do you want to continue?
```

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/ExamplesLine.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+import { LineStackLayout } from '../../../../UI/Layout';
+import FlatButton from '../../../../UI/FlatButton';
+import { Trans } from '@lingui/macro';
+import Add from '../../../../UI/CustomSvgIcons/Add';
+import ExamplesGrid from './ExamplesGrid';
+import { prepareExamples } from '../../../../AssetStore/ExampleStore';
+
+type ExamplesLineProps = {|
+  exampleShortHeaders: Array<ExampleShortHeader>,
+  onExpand: () => void,
```
Comment (4ian) [inline newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/ExamplesLine.js:15]: Ubert nit: onShowAll (to align with the name of the button)?

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/index.js [hunk 10]
```diff
@@ -261,6 +283,29 @@ const BuildSection = React.forwardRef<Props, BuildSectionInterface>(
...
+    const createProject = async (
+      i18n: I18nType,
+      settings: ProjectCreationSettings
+    ) => {
+      setIsOpeningProject(true);
+
+      try {
+        let projectMetadata;
+
+        projectMetadata = await onCreateBlank({
```
Comment (4ian) [inline newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/index.js:295]:
```suggestion
        const projectMetadata = await onCreateBlank({
```

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/index.js (hunk 4 above)
Comment (4ian) [reply newIDE/app/src/MainFrame/EditorContainers/HomePage/index.js:192]: Maybe
```suggestion
                      onShowAllExamples={() => onCreateProject(/*exampleShortHeader=*/null)}
```

to hint to the reader that this is opening the create project but without a select example?

**Review by 4ian — APPROVED** [review]
Added more comments, but I think this looks good once they are solved. Might be good to get a new review from @AlexandreSi though to be sure everything is clear.

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/ExamplesLine.js (hunk 2 above)
Comment (ClementPasteau) [reply newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/ExamplesLine.js:19]: you're totally right

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/ExamplesGrid.js (hunk 1 above)
Comment (ClementPasteau) [reply newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/ExamplesGrid.js:100]: yup, that makes sense

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/ExamplesLine.js (hunk 9 above)
Comment (ClementPasteau) [reply newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/ExamplesLine.js:15]: Yeah I thought that if we change the wording to "see all" one day, we'd be happy the code is not linked to the wording.
But I think showAll describes well the action too

**Comment by LuniMoon** [comment]
**Minor changes:**
[I deleted the name on the template thumbnails](https://www.figma.com/file/rY6g1YJW8gMEEoyh8CdqTV/%236-Home-(GDevelop)?node-id=1296%3A4051) (I just realised that the thumbnail has the name already so there's no need to repeat it)
> Capture d’écran 2022-08-24 à 15 16 16
>

.
I'm not convinced by the "flex-wrap" for the templates on smaller sizes.
I've just checked on Miro (which has templates too) and they scale down the thumbnails (they start with 8, and they start reducing to 6, and they finish with 3) -[here's the video for ref](https://app.usebubbles.com/tuEBHWhTyjXeH51a7rdLkY/miro-resizing)- I don't know if there's a similar system on the app

> Capture d’écran 2022-08-24 à 15 16 38


Ok for opening the pop-up when the user clicks "Start building directly" on the "Get started section".

On the "Build" section, it would be great if we could detach the second pop-up (the one with the template info) and open only this one when the user clicks the template.
[Miro opens the template directly when the user clicks on it](https://app.usebubbles.com/7sP6xXjPzEQVnMFBq3haMj/miro-template-2). I'm not entirely convinced on doing it like this since **our** game templates are not THAT standardised as diagram methodologies, so I think that opening the description instead (without the listing) could be a plausible solution for us.
That being said, I know that implementation wise COULD be tricky (to detach the second pop-up) and it is true that I don't have any data to prove my hypothesis. It's more of a hunch.

So: let's keep it that way for now (even if it feels odd) until we learn more about how templates are used, on which step of the creator journey, and what do they expect from them.

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/index.js (hunk 3 above)
Comment (ClementPasteau) [reply newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/index.js]: Weirdly I've tried removing this but there must be a typing issue somewhere and can't make it work without...
Wasted a lot of time trying to fix, but can't get my head around what's going on 🤔 
(for info, I didn't introduce this, this is the current behavior in place, so no risk to introduce a bug)

File: newIDE/app/src/ProjectCreation/CreateProjectDialog.js [hunk 11]
```diff
@@ -87,49 +78,16 @@ const CreateProjectDialog = ({
         id="create-blank-project-button"
         label={<Trans>Create a blank project</Trans>}
         primary
-        onClick={() => {
-          setSelectedExampleShortHeader(null);
-          setPreCreationDialogOpen(true);
-        }}
+        onClick={() =>
+          onOpenProjectPreCreationDialog(/*exampleShortHeader*/ null)
```
Comment (AlexandreSi) [inline newIDE/app/src/ProjectCreation/CreateProjectDialog.js:82]: Should we use this notation or use an object?

File: newIDE/app/src/UI/ImageTileGrid.js [hunk 12]
```diff
@@ -105,54 +80,76 @@ const useStylesForTile = makeStyles(theme =>
 const ImageOverlay = ({ text }) => (
```
Comment (AlexandreSi) [inline newIDE/app/src/UI/ImageTileGrid.js:80]: Could you type this mini component?

File: newIDE/app/src/UI/ImageTileGrid.js [hunk 13]
```diff
@@ -105,54 +80,76 @@ const useStylesForTile = makeStyles(theme =>
...
-      {text ? secondsToMinutesAndSeconds(text) : <Trans>Text</Trans>}
+      {text}
     </Typography>
   </div>
 );
 
-type TutorialsGridProps = {|
-  tutorials: Array<Tutorial>,
-  limit?: number,
+export type Widget = {|
```
Comment (AlexandreSi) [inline newIDE/app/src/UI/ImageTileGrid.js:88]: Should this be `ImageTileContent` or `ImageTileComponent`?

File: newIDE/app/src/UI/ImageTileGrid.js [hunk 14]
```diff
@@ -105,54 +80,76 @@ const useStylesForTile = makeStyles(theme =>
...
 |};
 
-const TutorialsGrid = ({ tutorials, limit }: TutorialsGridProps) => {
+type ImageTileGridProps = {|
+  items: Array<Widget>,
+  getColumnsFromWidth: (width: WidthType) => number,
+  getLimitFromWidth?: (width: WidthType) => number,
+|};
+
+const ImageTileGrid = ({
```
Comment (AlexandreSi) [inline newIDE/app/src/UI/ImageTileGrid.js:102]: Glad to see a new UI component!

File: newIDE/app/src/MainFrame/ElectronMainMenu.js [hunk 15]
```diff
@@ -91,9 +92,20 @@ const buildAndSendMenuTemplate = (
...
+        submenu: [
+          {
+            label: i18n._(t`New empty project...`),
+            accelerator: getElectronAccelerator(
+              shortcutMap['CREATE_NEW_PROJECT']
+            ),
+            onClickSendEvent: 'main-menu-create-blank',
+          },
+          {
+            label: i18n._(t`New project from template...`),
```
Comment (AlexandreSi) [inline newIDE/app/src/MainFrame/ElectronMainMenu.js:105]: Just a thought: are the ellipsis' needed?

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/LearnSection/index.js [hunk 16]
```diff
@@ -39,6 +45,29 @@ export const TUTORIAL_CATEGORY_TEXTS = {
   },
 };
 
+export const secondsToMinutesAndSeconds = (seconds: number) => {
```
Comment (AlexandreSi) [inline newIDE/app/src/MainFrame/EditorContainers/HomePage/LearnSection/index.js:48]: Could this go to DateDisplay.js or something similar?

**Review by AlexandreSi — APPROVED** [review]
Great job putting those components in common to have only one at the end, feels cleaner!
A few things here and there and off we go

File: newIDE/app/src/MainFrame/ElectronMainMenu.js (hunk 15 above)
Comment (ClementPasteau) [reply newIDE/app/src/MainFrame/ElectronMainMenu.js:105]: Not really, but I copied the current design which seems to be:

- If this action is the final one, then no ellipsis
- If there will be a following dialog, then ellipsis

I think it makes sense

File: newIDE/app/src/MainFrame/EditorContainers/HomePage/LearnSection/index.js (hunk 16 above)
Comment (ClementPasteau) [reply newIDE/app/src/MainFrame/EditorContainers/HomePage/LearnSection/index.js:48]: I didn't know about this file, will do

**Pull request merged**